janusgraph-core/src/main/java/org/janusgraph/graphdb/types/system/SystemTypeManager.java

```diff
@@ -33,26 +35,25 @@
 
     static {
         synchronized (SystemTypeManager.class) {
-            ImmutableMap.Builder<Long, SystemRelationType> idBuilder = ImmutableMap.builder();
-            ImmutableMap.Builder<String, SystemRelationType> nameBuilder = ImmutableMap.builder();
-            for (SystemRelationType et : new SystemRelationType[]{BaseKey.SchemaCategory, BaseKey.SchemaDefinitionDesc,
-                    BaseKey.SchemaDefinitionProperty, BaseKey.SchemaName, BaseKey.SchemaUpdateTime,
-                    BaseKey.VertexExists,
-                    BaseLabel.VertexLabelEdge, BaseLabel.SchemaDefinitionEdge,
-                    ImplicitKey.ID, ImplicitKey.JANUSGRAPHID, ImplicitKey.LABEL,
-                    ImplicitKey.KEY, ImplicitKey.VALUE, ImplicitKey.ADJACENT_ID,
-                    ImplicitKey.TIMESTAMP, ImplicitKey.TTL, ImplicitKey.VISIBILITY
-                }) {
+            SystemRelationType[] systemRelationTypes = new SystemRelationType[]{BaseKey.SchemaCategory, BaseKey.SchemaDefinitionDesc,
+                BaseKey.SchemaDefinitionProperty, BaseKey.SchemaName, BaseKey.SchemaUpdateTime,
+                BaseKey.VertexExists,
+                BaseLabel.VertexLabelEdge, BaseLabel.SchemaDefinitionEdge,
+                ImplicitKey.ID, ImplicitKey.JANUSGRAPHID, ImplicitKey.LABEL,
+                ImplicitKey.KEY, ImplicitKey.VALUE, ImplicitKey.ADJACENT_ID,
+                ImplicitKey.TIMESTAMP, ImplicitKey.TTL, ImplicitKey.VISIBILITY
+            };
+            Map<Long, SystemRelationType> idBuilder = new HashMap<>(systemRelationTypes.length);
+            Map<String, SystemRelationType> nameBuilder = new HashMap<>(systemRelationTypes.length);
+            for (SystemRelationType et : systemRelationTypes) {
                 idBuilder.put(et.longId(), et);
                 nameBuilder.put(et.name(),et);
             }
+            SYSTEM_TYPES_BY_ID = Collections.unmodifiableMap(idBuilder);
+            SYSTEM_TYPES_BY_NAME = Collections.unmodifiableMap(nameBuilder);
 
-            SYSTEM_TYPES_BY_ID = idBuilder.build();
-            SYSTEM_TYPES_BY_NAME = nameBuilder.build();
-
-
-            ADDITIONAL_RESERVED_NAMES = ImmutableSet.of(
-                "key", "vertex", "edge", "element", "property", "label");
+            ADDITIONAL_RESERVED_NAMES = Collections.unmodifiableSet(
+                Stream.of("key", "vertex", "edge", "element", "property", "label").collect(Collectors.toSet()));
```

Comment:
Isn't the part `.collect(Collectors.toSet()));` dupilcated?

Reply:
Not sure I correctly understand the question here. I don't see duplication here.
`Collections.unmodifiableSet` is wrapping some `Set` into an unmodifiable wrapper. It doesn't add values into a new set. So, there is no way to construct `unmodifiableSet` if we don't have an underlying set which should be wrapped.
`ImmutableSet.of` - creates a new set with specified elements. So it is Immutable.
`Collections.unmodifiableSet` - wraps an existing Set into unmodifiable Set. So, it is Mutable via underlying set but cannot be modified via unmodifiable set. As we are loosing a reference to an underlying set here, it means that this unmodifiable set becomes Immutable, because nobody will be able to change its value, as there is no access to underlying Set anymore.